kuma/search/tests/test_views.py

```diff
@@ -163,6 +163,14 @@ class PaginationSearchView(SearchView):
         response = view(request)
         assert response.data['pages'] == 2
 
+    def test_paginate_too_large(self):
+        response = self.client.get('/en-US/search', {'page': 1000},
+                                   HTTP_HOST=settings.WIKI_HOST)
+        # It's a bit odd that an invalid query string parameter
+        # causes a 404 Not Found but that's how the paginator for
+        # rest_framework works.
```

Comment:
Agreed..that is a bit odd.

Reply:
I agree too. An HTTP 400 (bad request) with some more detail in the explanation seems more appropriate.